numpy/core/src/multiarray/mapping.c

```diff
@@ -2479,13 +2480,18 @@ PyArray_MapIterCheckIndices(PyArrayMapIterObject *mit)
     int i;
     NPY_BEGIN_THREADS_DEF;
 
-    if (mit->size == 0) {
-        /* All indices got broadcast away, do *not* check as it always was */
+    intp_type = PyArray_DescrFromType(NPY_INTP);
+
+    if (NpyIter_GetIterSize(mit->outer) == 0) {
+        /*
+         * There are two reasons why `mit->size` can be 0, either the
+         * advanced indices have no elements (after broadcasting) or the
+         * subspace is empty. When the subspace is empty, we will check
+         * whether the indices are correct, otherwise not.
```

Comment:
```suggestion
         * When the outer iteration is empty, the indices broadcast to an
         * empty shape, and in this case we do not check if there are out
         * of bounds indices.
```

Reply:
Sure, that seems to capture it.

Perhaps worth a remark that the loop below checks the indices _pre_-broadcast, which is why we need this case.